pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -509,6 +510,8 @@ func (p *PriorityQueue) isPodWorthRequeuing(logger klog.Logger, pInfo *framework
 				}
 				hint = framework.Queue
 			}
+			metrics.QueueingHintExecutionDuration.WithLabelValues(hintfn.PluginName, event.Label, queueingHintToLabel(hint, err)).Observe(metrics.SinceInSeconds(start))
```

Comment:
Just recording this metric has the potential of increasing latency, especially for such a small operation (ideally).

Look what we did for PluginExecutionDuration https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/metrics/metric_recorder.go#L121

Maybe we need to do something similar. It basically discards the metric update if we can't keep up with flushing.

Reply:
Done.